Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/ProgressIndicators/ProgressIndicatorOrbsRotator.cs

```diff
@@ -0,0 +1,207 @@
+
+        [SerializeField]
+        private GameObject[] orbs = null;
+        
+        [SerializeField]
+        private TextMeshPro messageText = null;
+
+        [SerializeField]
+        private ProgressIndicatorState state = ProgressIndicatorState.Closed;
+        
+        public float RotationSpeedRawDegrees;
+        public float SpacingDegrees;
+        public float Acceleration;
+        public int Revolutions;
+        public bool TestStop = false;
+        public bool HasAnimationFinished = false;
+
+        private float progress;
+        private float timeElapsed;
+        private int deployedCount;
+        private bool timeUpdated;
+        private float[] angles;
+        private float timeSlice;
+        private float deg2rad = Mathf.PI / 180.0f;
+        private Renderer[] dots = null;
+        private bool stopRequested;
+        private float rotationWhenStopped;
+        private MaterialPropertyBlock[] propertyBlocks = null;
+
+        public async Task OpenAsync()
+        {
+            gameObject.SetActive(true);
+
+            state = ProgressIndicatorState.Opening;
+
+            StartOrbs();
+
+            await Task.Yield();
+
+            state = ProgressIndicatorState.Open;
+        }
+
+        public async Task CloseAsync()
+        {
+            state = ProgressIndicatorState.Closing;
+
+            StopOrbs();
+
+            while (!HasAnimationFinished)
+            {
+                await Task.Yield();
+            }
+
+            state = ProgressIndicatorState.Closed;
+
+            gameObject.SetActive(false);
+        }
+
+        private void StartOrbs()
+        {
+            rotationWhenStopped = 0.0f;
+            stopRequested = false;
+            timeSlice = 0;
+            timeUpdated = false;
+            deployedCount = 1;
+            timeElapsed = 0.0f;
+            angles = new float[orbs.Length];
+            for (int i = 0; i < angles.Length; ++i)
+            {
+                angles[i] = 0;
+            }
+
+            dots = new Renderer[5];
+            propertyBlocks = new MaterialPropertyBlock[dots.Length];
+            
+            for (int i = 0; i < orbs.Length; ++i)
+            {
+                propertyBlocks[i] = new MaterialPropertyBlock();
+                propertyBlocks[i].SetColor("_Color", new Color(1, 1, 1, 1));
+                dots[i] = orbs[i].transform.GetChild(0).gameObject.GetComponent<Renderer>();
+                dots[i].SetPropertyBlock(propertyBlocks[i]);
+            }
+        }
+
+        public void StopOrbs()
+        {
+            stopRequested = true;
+            rotationWhenStopped = angles[0];
+        }
+
+        private void Update()
+        {
+            if (state == ProgressIndicatorState.Closed)
+                return;
+
+            if (HasAnimationFinished == false)
+            {
+                UpdateTime();
+                ControlDotStarts();
+                IncrementOrbs();
+                HandleTestStop();
```

Comment:
Consider making HandleTestStop an editor-only thing (I assume it's not something that will be exercised in player mode)